src/pop/blockchain/alt_block_tree.cpp

```diff
@@ -468,10 +468,28 @@ AltBlockTree::BlockPayloadMutator::BlockPayloadMutator(tree_t& tree,
   appendIds<VbkBlock>(ids_, block_);
 }
 
-bool AltBlockTree::BlockPayloadMutator::isStatefulDuplicate(
-    const id_vector_t& payload_id) {
-  // make sure existing blocks do not contain this id
-  for (const auto& hash : payload_index_.getContainingAltBlocks(payload_id)) {
+template <>
+bool AltBlockTree::BlockPayloadMutator::isStatefulDuplicate<ATV>(
+    const ATV::id_t& atv_id) {
+  auto id = atv_id.asVector();
+  // make sure existing blocks from the active chain do not contain this id
+  for (const auto& hash : payload_index_.getContainingAltBlocks(id)) {
+    const auto* candidate = tree_.getBlockIndex(hash);
+    if (chain_.contains(candidate)) {
```

Comment:
For ATV we could see if "block of proof" is in VBK tree (not necessarily in active chain), if yes -> it will contain a list of IDs. If ID exists -> it's a duplicate.